git-credential-manager: 2.1.2 -> 2.2.2

josephst opened the pull request:
Update to 2.2.2, including bumping .NET version to 7.0 (to match upstream change)

See https://github.com/git-ecosystem/git-credential-manager/releases/tag/v2.2.0 for .NET 7 update details

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))

Note: build fails with `sandbox = true` on Darwin due to limitation with .NET running on MacOS - expects to be able to run `/usr/bin/codesign` to sign the binary to comply with MacOS security requirements, but running this binary doesn't work with sandbox enabled. 

- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

josephst committed:
git-credential-manager: 2.1.2 -> 2.2.2
Update to 2.2.2, including bump .NET version to 7.0 (to match upstream change)

See https://github.com/git-ecosystem/git-credential-manager/releases/tag/v2.2.0
for .NET 7 update details

Janik-Haag commented on pkgs/applications/version-management/git-credential-manager/default.nix:
```diff
@@ -7,38 +7,49 @@
+  };
 
   meta = with lib; {
     description = "Secure, cross-platform Git credential storage with authentication to GitHub, Azure Repos, and other popular Git hosting services.";
```
```suggestion
    description = "A secure, cross-platform Git credential storage with authentication to GitHub, Azure Repos, and other popular Git hosting services";
```
since we are already updating the package we can also fix the description so it fit's contributing.md

SuperSandro2000 replied:
We don't need the A there, it is just commonly used to replace the name of the package

Janik-Haag replied:
We still don't put a dot at the end of the description. I'm fine without the `A`.  And `A` isn't used to replace the name of the package since we don't allow the package name in the description attribute.

Janik-Haag commented on pkgs/applications/version-management/git-credential-manager/default.nix:
```diff
@@ -7,38 +7,49 @@
 , withGpgSupport ? true
 }:
 
+# note: requires sandbox to be disabled on MacOS, so that
+# .NET can find `/usr/bin/codesign` to sign the compiled binary
```
This should probably be added in the meta.longDescription attribute so users can actually find it since most people won't look at a package definition.

999eagle replied:
This sounds like it's a general issue on macOS affecting many packages but I'm personally not using macOS, so I can't check. Should we really add this comment to specifically this package?

Janik-Haag replied:
For me it sounds like a specific issue with `/usr/bin/codesign` on darwin so I would still add it to this package. But I don't even own a mac so I don't care too much tbh. And I think we can both agree that meta.longDescription is easier to find for users getting having this problem then some comment in the package definition?

999eagle replied:
`git-credential-manager` can't be the only package that requires code signing on macOS, especially as this isn't explicitly done anywhere. But yes, the `longDescription` would be a better place than a comment if we have to add it here.

Janik-Haag replied:
I'm in favor of adding it here so we can avoid user frustration, because it doesn't really hurt if we have it and in case it's not that probably not that unlikely that this is the  first time $user needs codesign.

reckenrode replied:
As far as I can tell, upstream is only doing codesigning in the GitHub action. I have the sandbox set to relaxed on Darwin, and this builds, but it crashes because it’s not signed at all. It should be possible to use darwin.sigtool’s `codesign` in a `postFixup` to manually ad hoc sign the resulting binary. 

Also, Is it possible to use `install_name_tool` to make it use libc++ from nixpkgs instead of the system one? I assume that’s an artifact of creating a self-contained binary?

Janik-Haag commented:
Result of `nixpkgs-review pr 245442` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>git-credential-manager</li>
  </ul>
</details>

Janik-Haag reviewed (commented):
LGTM; just two small nitpicks

999eagle commented on pkgs/applications/version-management/git-credential-manager/default.nix:
```diff
@@ -7,38 +7,49 @@
     ++ lib.optionals withGuiSupport [ libX11 libICE libSM ]
     ++ lib.optional withLibsecretSupport libsecret;
-  makeWrapperArgs = lib.optionals withGpgSupport [ "--prefix" "PATH" ":" (lib.makeBinPath [ gnupg pass ]) ];
+  makeWrapperArgs = lib.optionals withGpgSupport [ "--prefix" "PATH" ":" (lib.makeBinPath [ git gnupg pass ]) ];
```
`git` should probably be added to the path unconditionally, not just when gpg support is enabled.

josephst commented:
Updated `longDescription` with the sandbox comment and added `git` to the path. 

With respect to code signing on .NET - it could be fixed with a postFixup hook to sign the binary (possibly using something like [rcodesign](https://github.com/indygreg/apple-platform-rs) to avoid depending on MacOS-specific binaries). However, I think a better solution would be adding a similar postFixup hook to `buildDotnetModule` to sign binaries at build-time (with`sigtool` or `rcodesign`) which would fix this problem on all .NET packages on MacOS.

SuperSandro2000 commented on pkgs/applications/version-management/git-credential-manager/default.nix:
```diff
@@ -17,34 +20,48 @@
     license = with licenses; [ mit ];
     platforms = platforms.unix;
     maintainers = with maintainers; [ _999eagle ];
+    longDescription = ''
```
longDescription is normally placed directly under description
